src/Analysis/Codelyzer.Analysis.CSharp/CSharpRoslynProcessor.cs

```diff
@@ -309,6 +311,13 @@ private void AddChildNodes(UstList<UstNode> nodeChildren, SyntaxNode syntaxNode)
                 nodeChildren.AddRange(children);
             }
         }
+        private void AssignParentNode(List<UstNode> children, UstNode parentNode)
+        {
+            foreach (var child in children)
+            {
+                child.Parent = parentNode;
+            }
+        }
```

Comment:
1) Do we need a flag for this?
2) Is this part of the report? If yes, how does it look like?
3) How much does the size of the result increase when adding this?

Reply:
The `Parent` property is decorated with the `[JsonIgnore]` attribute. This excludes it from the report and prevents any impact on the file size.

Reply:
Great. That answers question 2, but can you check if the object being generated will be taking up significantly more memory? If yes, we might need a flag for this.

Reply:
Thanks for the clarification and good insight. I profiled a run on a larger project (NopCommerce) and found the memory impact to be negligible (in my opinion). Out of a total of 2.7 GB consumed, we saved ~1 MB by not including the Parent property. There is a small footprint here because we're not creating any new objects, we are just holding a reference from each child node.

Full stats of the comparison:
* Without Parent property:
  * 2.7 GB total memory consumed
  * 627.5 MB on the heap
  * 8,821,000 objects on the heap
* With Parent property:
  * 2.7 GB total memory consumed
  * 628.6 MB on the heap
  * 8,834,00 objects on the heap
  
I can add a flag if you think it is warranted, but my opinion is that the Parent property should always be included as a Parent is usually accessible from a child node in a tree, and the cost of maintaining that relationship is minimal.